src/lsp-socket.ts

```diff
@@ -2,10 +2,16 @@ import * as os from "node:os";
 import * as dns from "node:dns/promises";
 import { AnyARecord } from "node:dns";
 import * as net from "node:net";
-import { gotCliFlag } from "./cli-flags.js";
+import { getCliOption, gotCliFlag } from "./cli-flags.js";
 
 export async function getWindowsHostAddress() {
-  const useMirroredNetworking = gotCliFlag("--useMirroredNetworking");
+  const manualHost = getCliOption("host");
+  if (manualHost) {
+    console.log(`Using manually specified host: ${manualHost}`);
```

Comment:
stdout is reserved for communicating with the LSP client, use `logger.debug` instead